File: app/src/main/java/com/example/android/androidskeletonapp/ui/login/LoginResult.java
package com.example.android.androidskeletonapp.ui.login;

import androidx.annotation.Nullable;

import org.hisp.dhis.android.core.user.User;

class LoginResult {
    @Nullable
    private User success;

    @Nullable
    private String error;

    LoginResult(@Nullable String error) {
        this.error = error;
    }

    LoginResult(@Nullable User success) {
        this.success = success;
    }

    @Nullable
    User getSuccess() {
        return success;
    }

    @Nullable
    String getError() {
        return error;
    }
}

File: app/src/main/java/com/example/android/androidskeletonapp/ui/login/LoginResult.kt
package com.example.android.androidskeletonapp.ui.login

import org.hisp.dhis.android.core.user.User

internal class LoginResult {
    var success: User? = null
        private set
    var error: String? = null
        private set

    constructor(error: String?) {
        this.error = error
    }

    constructor(success: User?) {
        this.success = success
    }
}
